### PR TITLE
Firebase Analytics + Sentry error logging

### PR DESCRIPTION
**Pull request by wangannie**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR
* Integrated Firebase Analytics (new compatibility with Expo SDK 37 from #52 ) as a free alternative to FullStory. Getting a clear view of how clerks navigate the app can hopefully assist with remote user testing and expose bugs or points of confusion in the UX.
* Added Analytics logging for the following events:
  - ClerkLogOut
  - GoBackApplyRewards
  - ConfirmTransaction
  - ClerkLogin
  - CustomerFound
  - SelectTrainingMode
  - ConfirmQuantity
  - OpenQuantityModal
  - OpenRewardsModal
  - ConfirmApplyRewards
* Added Sentry error logging in all try/catch blocks

### Online sources
[Firebase setup](https://docs.expo.io/guides/setup-native-firebase/?redirected)
[Expo Firebase Analytics](https://docs.expo.io/versions/latest/sdk/firebase-analytics/)

### Related PRs
Analogous to https://github.com/calblueprint/dccentralkitchen/pull/88/

## How to review
* ~~In `App.js` set `  Analytics.setAnalyticsCollectionEnabled(false)` to `true`.~~
* Log into the [Firebase console](https://console.firebase.google.com/) (shared with healthyrewards@calblueprint.org , credentials pinned in Slack)
* In the console, go to `StreamView` under `Analytics` in the menu on the left
* Click around the app -- complete test transactions, clerk login, clerk training mode etc and see if the events are properly logged. You may need to refresh the Firebase console a few times for events to start showing up.
* Look through the Events and various parameters and make sure everything is accurate. Look out for anything that's marked `undefined`.
* Make sure the event names and parameters all make sense

## Next steps
* Possibly clean up the event logging into functions in `logUtils` like Sentry. This might be difficult to standardize since the parameters attached to each event vary so much, but 
* More extensive event logging--look into tracking each transaction process as a session instead of individual events

[//]: # "What's NOT in this PR, doesn't work yet, and/or still needs to be done"

## Tests Performed, Edge Cases
* Tested several transactions/combinations etc.

### Screenshots
* Successful event logging in the Firebase Console
analytics events

CC: @anniero98 

[//]: # 'This tags in both Annies as a default. Feel free to change, or add on anyone who you should be in on the conversation.'

### CODE REVIEW

**Comment by anniero98** [comment]
I keep getting
```
Possible Unhandled Promise Rejection (id 3): Error: The method or property expo-firebase-analytics.setAnalyticsEnabled is not available on ios, are you sure you've linked all the native dependencies properly?
```
(this is testing on iPad btw) nothing seems wrong, but I also installed packages yarn complained about just in case. Anyway unable to test it because of that 😅

**Comment by anniero98** [comment]
I'm getting a new error logging into A&S with `1111` ): 
```
Possible Unhandled Promise Rejection (id 0): Error: Invalid user-property value specified. Value should be a string of up to 36 characters long.
```

File: screens/modals/QuantityModal.js [hunk 1]
```diff
@@ -68,6 +69,13 @@ export default class QuantityModal extends React.Component {
     const currentQuantityInt = parseInt(this.state.currentQuantity, 10);
     const priceDifference = (currentQuantityInt - initialQuantity) * this.state.product.customerCost;
     this.props.callback(this.state.product, currentQuantityInt, priceDifference);
+    Analytics.logEvent('ConfirmQuantity', {
+      name: 'Apply product quantity',
+      function: 'handleUpdateCart',
+      screen: 'QuantityModal',
```
Comment (tommypoa) [inline screens/modals/QuantityModal.js:75]: It might be confusing to have `screen` as an event property since `screen_name` is captured with the Analytics too. Perhaps `component`?

Comment (wangannie) [reply screens/modals/QuantityModal.js]: Good idea! Thanks Tommy

File: screens/ClerkLoginScreen.js [hunk 2]
```diff
@@ -82,6 +83,19 @@ export default class ClerkLoginScreen extends React.Component {
...
+          clerk_name: clerkRecord.clerkName,
+          store: clerkRecord.storeName[0],
+        });
+        Analytics.logEvent('ClerkLogin', {
+          name: 'Successful Clerk login',
+          function: 'handleSubmit',
+          component: 'ClerkLoginScreen',
+          clerk_id: clerkRecord.id,
+          clerk_name: clerkRecord.clerkName,
+          store_name: clerkRecord.storeName[0],
```
[the remark this answers is not in the document]
Comment (wangannie) [reply screens/ClerkLoginScreen.js:97]: Yeah apparently they do return arrays. Not sure if there's a better way to do this, since clerks should only ever be connected to a single store?

File: screens/StoreLookupScreen.js [hunk 3]
```diff
@@ -69,6 +70,11 @@ export default class StoreLookupScreen extends React.Component {
   handleNavigate = async () => {
     // Clerk training: set `trainingMode` to "true" in AsyncStorage
     if (this.state.store.storeName === 'CLERK TRAINING') {
+      await Analytics.logEvent('SelectTrainingMode', {
```
[the remark this answers is not in the document]
Comment (wangannie) [reply screens/StoreLookupScreen.js:78]: Yeah you're right. I removed the `await`s
